systemtest/src/test/java/io/strimzi/systemtest/operators/RolesOnlyOperatorST.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.systemtest.operators;
+
+import io.strimzi.systemtest.AbstractST;
+import io.strimzi.systemtest.Environment;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.systemtest.resources.operator.BundleResource;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static io.strimzi.systemtest.Constants.REGRESSION;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+@Tag(REGRESSION)
+class RolesOnlyOperatorST extends AbstractST {
+
+    static final String NAMESPACE = "roles-only-cluster-test";
+    static final String CLUSTER_NAME = "roles-only-cluster";
+
+    private static final Logger LOGGER = LogManager.getLogger(RolesOnlyOperatorST.class);
+
+    @Test
+    void testRolesOnlyDeploysRoles() {
+        assumeTrue(Environment.isRolesOnly());
+        prepareEnvironment();
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 3, 3).done();
+
+        // Obviously not right.
+        // Get advice on how to resolve what this test should do.
+        assertThat(kubeClient().listClusterRoles(), is(Collections.emptyList()));
+    }
+
+    private void prepareEnvironment() {
+        prepareEnvForOperator(NAMESPACE);
+        applyBindings(NAMESPACE);
+        // 060-Deployment
+        BundleResource.clusterOperator(NAMESPACE).done();
+    }
+}
```

Comment:
I'm not sure I follow the point of this test. It seems to run only when the tests are triggered with `isRolesOnly == true`, but it doesn't really test anything else what the other tests would not do. So it seems completely useless to me (of course it is possible i miss the point). I think much bigger value would be if:
* This test runs in regular Acceptance pipeline
* It deploys the CO with the roles only setup
* Tests a simple deployment, topic and user creation and sending of messages

That would be IMHO added value.

Reply:
I think you're right...
I was thinking there was value in checking that a rolesOnly deployment had no cluster roles -this would stop new ClusterRoles sneaking into the deployment and 'breaking' the point of tihs mode, maybe youre right that ssimply installing in rolesOnly and then runnign acceptance is enough, but unfortunately this doesnt cover the case where someones adds a new ClusterRole and doesn't think about how that factors into this feature.